### DISCUSSION TITLE
why can't run `pnpm  run dev  --filter package`

### DISCUSSION THREAD

**Discussion by WingDust** [pnpm/pnpm#5938] [Q&A] [answered]
I am in windows 10 
use pnpm 7.24.3
run  `pnpm  run dev  --filter package`  get  error
```txt
 ERR_PNPM_NO_SCRIPT  Missing script: dev
Exception: pnpm exited with 1
```
but 
`pnpm --filter package run dev`  is work

**Comment by iobuhov** [comment, marked as answer]
It looks like putting arguments after `dev` script/command force pnpm to:
1. Search this script in local (current) package
2. To pass `--filter pacakge` as arguments to this script (as it may be options for `dev` command).

But if you use filter flag *before* `run` command, then it knows that this is `pnpm` options and run command correctly.
I'm always use filters right after `pnpm` to avoid this problems.

**Comment by liudonghua123** [comment]
I have the similar issue when run `pnpm build:copy` in [nocodb/packages/nc-gui](https://github.com/nocodb/nocodb/blob/develop/packages/nc-gui/package.json), and I noticed that the script name is appended with semicolon and the semicolon is not the same usage on windows.

```
Liu.D.H  nc-gui  git(nocodb-feat/oracle-support)   8ms  20:58 > pnpm build:copy

> nc-gui@ build:copy D:\code\node\nocodb\packages\nc-gui
> pnpm run generate; rm -rf ../nc-lib-gui/lib/dist/; rsync -rvzh ./dist/ ../nc-lib-gui/lib/dist/

 ERR_PNPM_NO_SCRIPT  Missing script: generate;

Command "generate;" not found. Did you mean "pnpm run generate"?
 ELIFECYCLE  Command failed with exit code 1.

Liu.D.H  nc-gui  git(nocodb-feat/oracle-support)   2.153s  20:58 >
```

To fix it, just replace `;` with `&`. 


See also https://woshub.com/run-multiple-commands-one-line-powershell-cmd/, https://www.baeldung.com/linux/difference-ampersand-semicolon, https://superuser.com/questions/345602/what-is-cmds-equivalent-to-bashs-ampersand-for-running-a-command-without-w.
